addons/web/static/src/views/fields/numpad_decimal_hook.js

```diff
@@ -35,19 +35,18 @@ export function useNumpadDecimal() {
             "end"
         );
     };
-    useEffect(
-        (el) => {
-            if (el) {
-                const inputs = el.nodeName === "INPUT" ? [el] : el.querySelectorAll("input");
-                inputs.forEach((input) => {
-                    listeners.push(input);
-                    input.addEventListener("keydown", handler);
-                });
-            }
-            return () => {
-                listeners.forEach((input) => input.removeEventListener("keydown", handler));
-            };
-        },
-        () => [ref.el]
-    );
+    useEffect(() => {
+        const el = ref.el;
+        if (el) {
+            const inputs = el.nodeName === "INPUT" ? [el] : el.querySelectorAll("input");
```

Comment:
Maybe you could do this to get rid of the listeners array ?
```suggestion
        let inputs = [];
        if (el) {
            inputs = el.nodeName === "INPUT" ? [el] : el.querySelectorAll("input");
```

If you want, you can also move the `let inputs = [];` in the closure of the useNumpadDecimal hook instead of the useEffect callback, I don't think that it matters a lot.